pkg/activator/activator.go

```diff
@@ -33,3 +39,27 @@ var (
 		RevisionHeaderNamespace,
 	}
 )
+
+// GetNameAndNamespace If the headers aren't explicitly specified,
+// then decode the revision name and namespace from the Host header.
+func GetNameAndNamespace(name, namespace, host string) (string, string) {
+	if name == "" || namespace == "" {
+		parts := strings.SplitN(host, ".", 4)
+		switch len(parts) {
```

Comment:
I think this switch would match `foo.bar.svc` and `foo.bar`, which isn't the case in the original code. Are we intending to widen the number of formats we accept here?